syweb/webclient/app/settings/settings-controller.js

```diff
@@ -455,4 +458,50 @@ function($scope, matrixService, modelService, eventHandlerService, notificationS
             dialogService.showError(err);
         });
     };
+
+    $scope.passwordValid = function() {
+        if ($scope.password.oldpw == '') {
+            return false;
+        }
+        if ($scope.password.newpw == '') {
+            return false;
+        }
+        if ($scope.password.confirmnewpw == '') {
+            return false;
+        }
+        return true;
+    };
+
+    $scope.changePassword = function() {
+        if ($scope.password.newpw != $scope.password.confirmnewpw) {
```

Comment:
Only problem with this is that if one of the password boxes is blank, it's fairly obvious why the button is greyed out. Theres no equivalent visual cue for a password mismatch. To do this properly we'd have to make the validation give hints as to what was wrong.